etcd3/locks.py

```diff
@@ -55,54 +55,61 @@ def acquire(self, timeout=10):
         :returns: True if the lock has been acquired, False otherwise.
 
         """
-        stop = (
-            tenacity.stop_never
-            if timeout is None else tenacity.stop_after_delay(timeout)
-        )
+        if timeout is not None:
+            deadline = time.time() + timeout
 
-        def wait(previous_attempt_number, delay_since_first_attempt):
-            if timeout is None:
-                remaining_timeout = None
-            else:
-                remaining_timeout = max(timeout - delay_since_first_attempt, 0)
-            # TODO(jd): Wait for a DELETE event to happen: that'd mean the lock
-            # has been released, rather than retrying on PUT events too
-            try:
-                self.etcd_client.watch_once(self.key, remaining_timeout)
-            except exceptions.WatchTimedOut:
-                pass
-            return 0
-
-        @tenacity.retry(retry=tenacity.retry_never,
-                        stop=stop,
-                        wait=wait)
-        def _acquire():
-            # TODO: save the created revision so we can check it later to make
-            # sure we still have the lock
```

Comment:
This is partly done. Revision in saved as instance variable. I don't know why it is needed to check if the lock is acquired. UUID comparison is done for this purpose.